File: Kernel Simulator/TestShell/Commands/SleepTook.vb

'    Kernel Simulator  Copyright (C) 2018-2022  Aptivi
'
'    This file is part of Kernel Simulator
'
'    Kernel Simulator is free software: you can redistribute it and/or modify
'    it under the terms of the GNU General Public License as published by
'    the Free Software Foundation, either version 3 of the License, or
'    (at your option) any later version.
'
'    Kernel Simulator is distributed in the hope that it will be useful,
'    but WITHOUT ANY WARRANTY; without even the implied warranty of
'    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
'    GNU General Public License for more details.
'
'    You should have received a copy of the GNU General Public License
'    along with this program.  If not, see <https://www.gnu.org/licenses/>.

Namespace TestShell.Commands
    Class Test_SleepTookCommand
        Inherits CommandExecutor
        Implements ICommand

        Public Overrides Sub Execute(StringArgs As String, ListArgs() As String, ListArgsOnly As String(), ListSwitchesOnly As String()) Implements ICommand.Execute
            Dim SleepMs As Integer = ListArgsOnly(0)
            Dim Ticks As Boolean = ListSwitchesOnly.Contains("-t")
            If Ticks Then
                Write("{0} ms => {1} ticks", True, color:=GetConsoleColor(ColTypes.Neutral), SleepMs, GetActualTicks(SleepMs))
            Else
                Write("{0} ms => {1} ms", True, color:=GetConsoleColor(ColTypes.Neutral), SleepMs, GetActualMilliseconds(SleepMs))
            End If
        End Sub

        Public Overrides Sub HelpHelper()
            Write(DoTranslation("This command has the below switches that change how it works:"), True, GetConsoleColor(ColTypes.Neutral))
            Write("  -t: ", False, GetConsoleColor(ColTypes.ListEntry)) : Write(DoTranslation("Unit in ticks"), True, GetConsoleColor(ColTypes.ListValue))
        End Sub

    End Class
End Namespace

File: Kernel Simulator/TestShell/Commands/SleepTook.cs
using System;
using System.Linq;
using KS.ConsoleBase.Colors;
using KS.Languages;
using KS.Misc.Threading;
using KS.Misc.Writers.ConsoleWriters;
using KS.Shell.ShellBase.Commands;
using Microsoft.VisualBasic.CompilerServices;

// Kernel Simulator  Copyright (C) 2018-2022  Aptivi
// 
// This file is part of Kernel Simulator
// 
// Kernel Simulator is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// Kernel Simulator is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.

namespace KS.TestShell.Commands
{
	class Test_SleepTookCommand : CommandExecutor, ICommand
	{

		public override void Execute(string StringArgs, string[] ListArgs, string[] ListArgsOnly, string[] ListSwitchesOnly)
		{
			int SleepMs = Conversions.ToInteger(ListArgsOnly[0]);
			bool Ticks = ListSwitchesOnly.Contains("-t");
			if (Ticks)
			{
				TextWriterColor.Write("{0} ms => {1} ticks", true, color: KernelColorTools.GetConsoleColor(KernelColorTools.ColTypes.Neutral), SleepMs, ThreadManager.GetActualTicks(SleepMs));
			}
			else
			{
				TextWriterColor.Write("{0} ms => {1} ms", true, color: KernelColorTools.GetConsoleColor(KernelColorTools.ColTypes.Neutral), SleepMs, ThreadManager.GetActualMilliseconds(SleepMs));
			}
		}

		public override void HelpHelper()
		{
			TextWriterColor.Write(Translate.DoTranslation("This command has the below switches that change how it works:"), true, KernelColorTools.GetConsoleColor(KernelColorTools.ColTypes.Neutral));
			TextWriterColor.Write("  -t: ", false, KernelColorTools.GetConsoleColor(KernelColorTools.ColTypes.ListEntry));
			TextWriterColor.Write(Translate.DoTranslation("Unit in ticks"), true, KernelColorTools.GetConsoleColor(KernelColorTools.ColTypes.ListValue));
		}

	}
}